otlp/trace.go

```diff
@@ -35,25 +35,45 @@ type Event struct {
 }
 
 func TranslateTraceRequestFromReader(body io.ReadCloser, ri RequestInfo) (*TranslateTraceRequestResult, error) {
-	if err := ri.ValidateHeaders(); err != nil {
+	if err := ri.ValidateTracesHeaders(); err != nil {
 		return nil, err
 	}
 	request, err := parseOTLPBody(body, ri.ContentEncoding)
 	if err != nil {
 		return nil, ErrFailedParseBody
 	}
-	return TranslateTraceRequest(request)
+	return TranslateTraceRequest(request, ri)
 }
 
-func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest) (*TranslateTraceRequestResult, error) {
-	batch := []Event{}
+func TranslateTraceRequest(request *collectorTrace.ExportTraceServiceRequest, ri RequestInfo) (*TranslateTraceRequestResult, error) {
+	if err := ri.ValidateTracesHeaders(); err != nil {
+		return nil, err
+	}
```

Comment:
I didn't include header validation here before because our currently usage in refinery, Basenji & Shepherd wanted to validate headers before trying to translate the request.

Looking back, now I think it's fine to expect these funcs to perform the validation and consumers of Husky don't need to check themselves and instead expect validation errors to be returned from here.